src/cli/rustup_mode.rs

```diff
@@ -1594,18 +1594,14 @@ fn doc(cfg: &Cfg, m: &ArgMatches) -> Result<utils::ExitCode> {
     };
 
     let topical_path: PathBuf;
+    let doc_name = m.get_one::<String>("topic");
 
-    let doc_url = if let Some(topic) = m.get_one::<String>("topic") {
+    let doc_url = if let Some(topic) = doc_name {
         topical_path = topical_doc::local_path(&toolchain.doc_path("").unwrap(), topic)?;
         topical_path.to_str().unwrap()
-    } else if let Some((name, _, path)) = DOCS_DATA.iter().find(|(name, _, _)| m.get_flag(name)) {
-        writeln!(
-            process().stderr().lock(),
-            "Opening docs named `{name}` in your browser"
-        )?;
+    } else if let Some((_, _, path)) = DOCS_DATA.iter().find(|(name, _, _)| m.get_flag(name)) {
```

Comment:
```suggestion
    } else if let Some((name, _, path)) = DOCS_DATA.iter().find(|(name, _, _)| m.get_flag(name)) {
        doc_name = Some(name);
```

Reply:
is this `name` can be different from the provided one?

Reply:
> is this `name` can be different from the provided one?

Sorry, I don't quite see what you mean.
Anyway, in the case of `rustup doc --cargo`, `name` should be assigned "cargo".

Reply:
@0x61nas If you mean if `None` is ever possible, then the answer would be yes, but for testing purposes mostly.

Reply:
oh, i didnt see that, appreciate your response, thanks